public/pages/Main/Main.tsx

```diff
@@ -32,62 +44,155 @@ enum Pathname {
   Channels = '/channels',
 }
 
-interface MainProps extends RouteComponentProps {}
+interface MainProps extends RouteComponentProps {
+  setActionMenu: (menuMount: MountPoint | undefined) => void;
+  multiDataSourceEnabled: boolean;
+  dataSourceManagement: DataSourceManagementPluginSetup;
+}
 
-export interface MainState {
+export interface MainState extends Pick<DataSourceMenuProperties, "dataSourceId" | "dataSourceLabel"> {
   availableChannels: Partial<typeof CHANNEL_TYPE>;
   availableConfigTypes: string[]; // available backend config types
   tooltipSupport: boolean; // if true, IAM role for SNS is optional and helper text should be available
+  dataSourceReadOnly: boolean;
+  dataSourceLabel: string;
 }
 
 export const MainContext = createContext<MainState | null>(null);
 
 export default class Main extends Component<MainProps, MainState> {
   static contextType = ServicesContext;
-
   constructor(props: MainProps) {
     super(props);
-    this.state = {
-      availableChannels: CHANNEL_TYPE,
-      availableConfigTypes: [],
-      tooltipSupport: false,
-    };
+    let dataSourceId = "";
```

Comment:
setting the data source id as empty string actually makes the data source component think we are going to choose local cluster as the first option rather than the default data source, we can follow https://github.com/opensearch-project/index-management-dashboards-plugin/pull/1042/files#diff-6895fefa4e6e2c468c6231f4a896af1afd57803ceb66d3e207842debec9743c6R367, to add a loading status check, and before everything is loaded, the actionOption passed to data source picker should be undefined until the picker is mounted and passes back the data source id via callback